frontend/src/classes/MessageChain.ts

```diff
@@ -85,7 +85,8 @@ export default class MessageChain {
   /**
    * sets number of unviewed messages to zero
    */
-  resetNumberUnviewed(): void {
+  resetNumberUnviewed(): MessageChain {
     this._numberUnviewed = 0;
+    return this;
```

Comment:
Just curious- why does the MessageChain need to be returned?

Reply:
just makes the code cleaner! when app state gets changed, for example, app state is set as follows:
```
nextState.townMessageChain = [NEW VALUE]
```
without returning `this` we'd have to do:
```
const currentMessageChain = nextState.townMessageChain;
currentMessageChain.resetNumberUnviewed()
nextState.townMessageChain = currentMessageChain
```
instead of:
```
nextState.townMessageChain = nextState.townMessageChain.resetNumberUnviewed()
```
the latter is just easier to read